docs/modules/nifi/pages/usage_guide/custom_processors.adoc

```diff
@@ -0,0 +1,125 @@
+= Loading Custom Components
+
+It is possible to develop https://nifi.apache.org/docs/nifi-docs/html/developer-guide.html#introduction[custom components] for Apache NiFi to extend the available functionality - most often this will probably be custom processors.
+
+For these to be available in the NiFi UI, they need to be present in the classpath at startup, so the nar bundles have to be injected into your NiFi instance.
+
+The Stackable Data Platform supports two ways to achive this goal, both of which are described below.
+
+== Custom Docker Image
+
+You can extend the official Stackable NiFi image and simply copy all required nar files into the classpath of Nifi.
+The benefit of this method is that there is no need for any config overrides or extra mounts, you can simply use any ouf our NiFi examples, swap the image and your components will be available.
+But this means you will need to have access to a registry to push your custom image to.
+
+A simple Dockerfile would look like show in the following listing.
+
+[source,Dockerfile]
+----
+FROM docker.stackable.tech/stackable/nifi:1.21.0-stackable0.0.0-dev
+COPY /path/to/your/nar.file /stackable/nifi/lib/
+----
+
+You then need to make this image available to your Kubernetes cluster and specify it in your NiFi resource as described in xref:home:concepts:product_image_selection.adoc[].
+
+[source,yaml]
+----
+spec:
+  image:
+    productVersion: 1.21.0
+    custom: "docker.company.org/nifi:1.21.0-customprocessor"
+----
+
+== Using the Official Image
+If you don't want to create a custom image, then you can use the extra volume mount functionality to load components and configure NiFi with to read these from the extra volumes.
+
+For this to work you'll need to prepare a PVC containing your components.
+Usually the best way to do this will be to mount the PVC into a temporary container and then simply `kubectl cp` the nar files into that volume.
+
+The following listing shows an example of creating the PVC and the pod:
+
+[source, yaml]
+----
+---
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: nifi-processor
+spec:
+  accessModes:
+    - ReadWriteOnce
```

Comment:
Hmm, I think this solution should only be used as a last resort (we should clearly prefer the first one in the docs).
AFAIK `ReadWriteOnce` will only work for a single Node, so no HA setup using this method (and a single volume).
So I would change the docs to `ReadWriteMany` and state something like

If your Kubernetes installation does not support pvc with `ReadWriteMany` access mode, use `ReadWriteOnce` instead. However note, that this restricts you to a single NiFi instance. If this is not acceptable, please use the solution with a custim image

Reply:
That is completely true and is an ommission on my part.
I was intending to use `readonlymany` as access mode, which should be enough to load the processor and allow multiple pods while being more broadly available than `readwritemany`.

Reply:
`readonlymany` makes sense :+1:

Reply:
Yeah, but .... you cannot get data into the volume with readonlymany, because its ... read only .. and accessmode cannot be changed after the fact.
So basically the workflow would probably be, create a readwriteonce pvc, copy file in there, remove pvc, recreate with readonlymany and manually bind it to the pre-seeded PV.

Beginning to be less attractive I have to say..

@nightkr or is there any easier workaround to get a readonlymany volume claim with data?

Reply:
https://kubernetes.io/docs/concepts/storage/persistent-volumes/#access-modes
there is also `ReadWriteOncePod`, but I think I would simply go with `ReadWriteMany`

Reply:
You can create the PVC with multiple allowed access modes, as long as you only mount it into pods with one of them at a time.

So create the PVC with `accessModes: [ReadWriteOnce, ReadOnlyMany]`, mount it into your populating job with `ReadWriteOnce`, then once done mount it into your NiFis with `ReadOnlyMany`.

That said, it's still a pretty annoying workflow, and not really one that's compatible with Helm et al. :/

Reply:
We have an issue somewhere about being able to not just add mounts as external data but also urls or git repos.
That make it a bit nicer to my mind at least.

Reply:
>   Warning  ProvisioningFailed    13s (x3 over 58s)  rancher.io/local-path_local-path-provisioner-684f458cdd-22t4c_
> bc537abd-41cf-4fe7-b758-7664cf8018f5  failed to provision volume with StorageClass "standard": Only support ReadWr
> iteOnce access mode

All of this is so annoying ..

Reply:
Yes, local storage doesn't really make sense to use with `Read*Many` modes.